helmclient.go

```diff
@@ -118,7 +119,9 @@ func New(config Config) (*Client, error) {
 }
 
 // DeleteRelease uninstalls a chart given its release name.
-func (c *Client) DeleteRelease(ctx context.Context, releaseName string, options ...helmclient.DeleteOption) error {
+func (c *Client) DeleteRelease(ctx context.Context, releaseName string, options ...helmclient.DeleteOption) (err error) {
```

Comment:
I named on returning `error` variable so we can check it whether it's nil or not when we exit this function.

Reply:
I still don't get why. AFAICT this should be used only for documentation purposes.

I also don't think `errorReporting` is reliable. I already see some errors redeclared in backoff functions for example.

I'd rather go with `deleteRelease` called by `DeleteRelease` and check for errors there.